composer/models/resnet/model.py

```diff
@@ -76,9 +82,27 @@ def composer_resnet(model_name: str,
     if initializers is None:
         initializers = []
 
+    # Configure pretrained/weights based on torchvision version
+    if pretrained and weights:
+        raise ValueError(
+            'composer_resnet expects only one of ``pretrained`` or ``weights`` to be specified, but both were specified.'
+        )
+    if pretrained:
+        weights = 'IMAGENET1K_V2'
+        warnings.warn(
+            DeprecationWarning(
+                'The ``pretrained`` argument for composer_resnet is deprecated and will be removed in the future. Please use ``weights`` instead.'
+            ))
+    if weights:
```

Comment:
Moved into check. I'm in favor of emitting a warning which indicates its falling back to `pretrained=True` and that user should upgrade `torchvision` version to use `weights`. 

> Also, if weights and weights is not True (i.e. it is a string / path), then would like to raise an exception, as we wouldn't be able to load a customized weights path.

I don't understand what exactly you mean here. It can be a string argument -- see https://pytorch.org/vision/stable/models.html#initializing-pre-trained-models